src/main/java/org/spongepowered/api/Engine.java

```diff
@@ -25,12 +25,16 @@
 package org.spongepowered.api;
 
 import org.spongepowered.api.client.Client;
+import org.spongepowered.api.resource.ResourceManager;
+import org.spongepowered.api.resource.pack.PackRepository;
 import org.spongepowered.api.scheduler.Scheduler;
 
+import java.util.concurrent.Executor;
+
 /**
  * Shared functionality between {@link Client} and {@link Server} engines.
  */
-public interface Engine {
+public interface Engine extends Executor {
```

Comment:
You can't make engine a executor, all tasks need to be scheduled through the scheduler to know which plugin submitted them. You can however get a exectutor from the scheduler for a specific plugin.

Reply:
Minecraft and MinecraftServer are already executors. Adding a separate method would essentially be returning this. But I guess that's an implementation detail.

Reply:
We need to track which plugin posts a task, vanilla mc does not. So we can't just expose this in the API.